examples/virtual_methods/pet.rs

```diff
@@ -169,10 +175,13 @@ pub trait PetImplExt: PetImpl {
 }
 
 /// The `PetImplExt` trait is implemented for all subclasses that have [`Pet`] in the class hierarchy
-impl<T: PetImpl> PetImplExt for T {}
+impl<T: PetImpl> PetImplExt for T where <Self as ObjectSubclass>::Type: IsA<Pet> {}
```

Comment:
Not necessary (because nothing else can implement this trait) unless the compiler forces you :) It should also be implied via `T: PetImpl` already